microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/AcceptHeadersAndFormatParameterInteractionTest.java

```diff
@@ -0,0 +1,365 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+/**
+ * Test cases to verify negative scenario in which "Accept" HTTP header and {@code format} query parameter are
+ * conflicting or not valid.
+ * {@link MediaType} constants are used as HTTP "Accept" header values.
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({ AcceptHeadersAndFormatParameterInteractionTest.OpenApiExtensionSetup.class })
+@RunAsClient
+public class AcceptHeadersAndFormatParameterInteractionTest {
+
+    private final static String DEPLOYMENT_NAME = OpenApi10OnJaxRsAnnotationsTest.class.getSimpleName();
+
+    static OnlineManagementClient onlineManagementClient;
+
+    @Deployment(testable = false)
+    public static Archive<?> createDeployment() {
+        WebArchive deployment = ShrinkWrap.create(
+                WebArchive.class,
+                String.format("%s.war", DEPLOYMENT_NAME))
+                .addClasses(ProviderApplication.class)
+                .addClasses(
+                        District.class,
+                        DistrictEntity.class,
+                        DistrictService.class,
+                        InMemoryDistrictService.class,
+                        DistrictsResource.class,
+                        RoutingServiceConstants.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return deployment;
+    }
+
+    static class OpenApiExtensionSetup implements ServerSetupTask {
+
+        @Override
+        public void setup(ManagementClient managementClient, String containerId) throws Exception {
+            //  MP OpenAPI up
+            onlineManagementClient = ManagementClientProvider.onlineStandalone();
+            OpenApiServerConfiguration.enableOpenApi(onlineManagementClient);
+        }
+
+        @Override
+        public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
+            //  MP OpenAPI down
+            try {
+                OpenApiServerConfiguration.disableOpenApi(onlineManagementClient);
+            } finally {
+                onlineManagementClient.close();
+            }
+        }
+    }
+
+    /**
+     * @tpTestDetails Sends HTTP request with {@link MediaType#APPLICATION_JSON} {@code Accept} header and no {@code format}
+     *                query parameter.
+     * @tpPassCrit Despite missing {@code format} query parameter, the right Content Type is returned as per
+     *             MicroProfile OpenAPI spec.
+     *
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void testResponseJsonContentTypeWhenAcceptHeaderAndNoQueryParameter(@ArquillianResource URL baseURL)
+            throws URISyntaxException {
+        given().header("Accept", MediaType.APPLICATION_JSON).get(baseURL.toURI().resolve("/openapi"))
+                .then()
+                .statusCode(200)
+                .contentType(MediaType.APPLICATION_JSON);
```

Comment:
Could you update code formatting as it's in [1], please? It will unite formatting for MP OpenAPI tests and looks nicer to me :-)

[1] https://github.com/jboss-eap-qe/eap-microprofile-test-suite/blob/master/microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/v11/MicroProfileOpenApi11Test.java#L99

Reply:
Sure, will do.